framework/0019-edtf-formatting.md

```diff
@@ -0,0 +1,261 @@
+
+
+#### Formats for EDTF date intervals
+
+Date intervals are tricky because they can be formatted in multiple ways. Consider e.g. the interval ``2020-02/2020-03``. This interval could be formatted as below:
+
+```
+Naive: Feb 2020 - Mar 2020
+Compact: Feb-Mar 2020
+```
+
+**Naive approach**
+The first longer format is a single format, that can format the start/end EDTF date individually and merge them with an interval format.
+
+**Compact approach**
+The second compact format, requires an algorithm capable of determinig the *greatest difference* between a start and end date. For more information see [Common Locale Data Repository](https://www.unicode.org/reports/tr35/tr35-dates.html#intervalFormats)
+
+**Naive vs compact**
+As an initial implementation we could go with the naive approach, as long as, we can later implement the compact approach. In the naive approach we rely on an already defined fallback pattern per locale (used in ``babel.dates.format_interval``), and thus we don't need any further formats.
+
+
+### Python API
+
+We suggest to implement a new method ``format_edtf()`` that can format both EDTF date and date intervals.
+
+Examples:
+
+```python
+>>> d = parse_edtf('1939-09/1945')
+>>> format_edtf(d, format='medium', locale='en')
+Sep, 1939 - 1945
+```
+
+
+```python
+>>> d = parse_edtf('1939-09')
+>>> format_edtf(d, format='medium', locale='en')
+Sep, 1939
+```
+
+```python
+>>> d = parse_edtf('1939-09-01')
+>>> format_edtf(d, format='medium', locale='en')
+Sep 1, 1939
+```
+
+Format EDTF should also allow Python date objects (by simply falling back to ``format_date``):
+
+```python
+>>> d = date(1939, 9, 1)
+>>> format_edtf(d, format='medium', locale='en')
+Sep 1, 1939
+```
+
+**Flask and Babel**
+
+We will need two ``format_edtf`` functions. One pure utility function, and one which is Flask request context aware and able to determine use the current locale.
+
+#### Jinja template filters
+
+We also need a Jinja template filter to format dates:
+
+```
+{{my_edtfdate|edtfformat(format='long')}}
+```
+
+As well as a converter filter:
+
+
+```
+{{my_edtfdate_str|toedtf|edtfformat(format='long')}}
+```
+
+#### Marshmallow validation
+
+A Marshmallow field that uses the EDTF library needs to be built that are able to validate and parse EDTF dates. The field could possibly allow to specify if:
+
+- dates are allowed and/or
+- date intervals are allowed.
+
+#### Implementation
+
+We recommend implementing the pure utility function in a new module Babel-EDTF. The remaining part can be implemented in Invenio-I18N. This ensures that the core formatting capabilities are usable independently of Invenio.
+
+### Special considerations
+
+#### REST API
+
+We propose to create a Marshmallow field that can dump a localised EDTF date so that a client application will not need to know how to format an EDTF date, but can simply rely on the REST API to do it for them. This way we also don't need to build the formatting for the client-side.
```

Comment:
I am not sure I understand this part, in which format should it be dumped? I would just add more details here.